internal/feather.go

```diff
@@ -176,31 +178,36 @@ func setBuilderValues(builder array.Builder, slice reflect.Value, valid []bool)
 	case reflect.Bool:
 		builder.(*array.BooleanBuilder).AppendValues(values.([]bool), valid)
 	case reflect.Slice:
-		var offsets []int64
-		innerSliceType := slice.Type().Elem()
-		if innerSliceType.Kind() != reflect.Slice {
-			return errors.Errorf(
-				"expected slice of slices, instead found slice of '%s'",
-				innerSliceType.Kind(),
-			)
-		}
-		flatSlice := reflect.MakeSlice(innerSliceType, 0, slice.Len())
-		highwaterMark := 0
-		offsets = append(offsets, 0)
-		for i := 0; i < slice.Len(); i++ {
-			innerSlice := slice.Index(i)
-			highwaterMark += innerSlice.Len()
-			offsets = append(offsets, int64(highwaterMark))
-			flatSlice = reflect.AppendSlice(flatSlice, innerSlice)
-		}
-		if err := setBuilderValues(
-			builder.(*array.LargeListBuilder).ValueBuilder(),
-			flatSlice,
-			allValid(flatSlice.Len()),
-		); err != nil {
-			return errors.Wrap(err, "failed to set values for slice of slices")
+		// This is a byte slice, build a Binary array
+		if elemType.Elem().Kind() == reflect.Uint8 {
+			builder.(*array.BinaryBuilder).AppendValues(values.([][]byte), valid)
+		} else {
+			var offsets []int64
+			innerSliceType := slice.Type().Elem()
+			if innerSliceType.Kind() != reflect.Slice {
+				return errors.Errorf(
+					"expected slice of slices, instead found slice of '%s'",
+					innerSliceType.Kind(),
+				)
+			}
+			flatSlice := reflect.MakeSlice(innerSliceType, 0, slice.Len())
+			highwaterMark := 0
+			offsets = append(offsets, 0)
+			for i := 0; i < slice.Len(); i++ {
+				innerSlice := slice.Index(i)
+				highwaterMark += innerSlice.Len()
+				offsets = append(offsets, int64(highwaterMark))
+				flatSlice = reflect.AppendSlice(flatSlice, innerSlice)
+			}
+			if err := setBuilderValues(
+				builder.(*array.LargeListBuilder).ValueBuilder(),
+				flatSlice,
+				allValid(flatSlice.Len()),
+			); err != nil {
+				return errors.Wrap(err, "failed to set values for slice of slices")
+			}
+			builder.(*array.LargeListBuilder).AppendValues(offsets, valid)
```

Comment:
justa nna confirm that we control that this is a large list builder and not say a list builder

Reply:
yes, thank you!